### PR TITLE
DOC: Added an example for np.transpose(4d_array)

### PR DESCRIPTION
**Pull request by ruppysuppy**
See #16497

### CODE REVIEW

**Comment by eric-wieser** [comment]
Thanks for the contribution. This should be in the docs for transpose though, not the docs for einsum.

Also, the array you're testing with here is 2d not 4d

**Comment by ruppysuppy** [comment]
> Thanks for the contribution. This should be in the docs for transpose though, not the docs for einsum.
> 
> Also, the array you're testing with here is 2d not 4d

Yeah, a silly mistake on my part (using 2d).

Do I have to add the example in [numpy/doc/internals.py](https://github.com/numpy/numpy/blob/b7c27bd2a3817f59c84b004b87bba5db57d9a9b0/numpy/doc/internals.py)?

**Comment by eric-wieser** [comment]
> Do I have to add the example in numpy/doc/internals.py?

No, that's not where the docs for the transpose function are. You're looking for `numpy/core/fromnumeric.py`.

**Comment by ruppysuppy** [comment]
> > Do I have to add the example in numpy/doc/internals.py?
> 
> No, that's not where the docs for the transpose function are. You're looking for `numpy/core/fromnumeric.py`.

Thank you, I'll start working on it

**Comment by ruppysuppy** [comment]
It seems like it failed some tests, but I modified the documentation only

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -1693,6 +1694,11 @@
     array([[0, 3],
            [1, 4],
            [2, 5]])
+    >>> np.transpose(d)
+    array([[ 0,  4,  8, 12],
+           [ 1,  5,  9, 13],
+           [ 2,  6, 10, 14],
+           [ 3,  7, 11, 15]])
```
Comment (eric-wieser) [inline numpy/core/_add_newdocs.py:1701]: Let's undo the changes in this file, they don't add anything.

File: numpy/core/fromnumeric.py [hunk 2]
```diff
@@ -649,6 +649,49 @@ def transpose(a, axes=None):
     >>> np.transpose(x, (1, 0, 2)).shape
     (2, 1, 3)
 
+    >>> x = np.arange(24).reshape(2, 2, 2, 3)
```
Comment (eric-wieser) [inline numpy/core/fromnumeric.py:652]: A shape like `(2, 3, 4, 5)` would be more instructive I think

File: numpy/core/fromnumeric.py [hunk 3]
```diff
@@ -649,6 +649,49 @@ def transpose(a, axes=None):
...
+
+            [[ 4, 16],
+             [10, 22]]],
+
+
+           [[[ 2, 14],
+             [ 8, 20]],
+
+            [[ 5, 17],
+             [11, 23]]]])
```
Comment (eric-wieser) [inline numpy/core/fromnumeric.py:693]: Showing the arrays is just noise here IMO. This would be clearer as:

```suggestion
    >>> np.transpose(x).shape
    (3, 2, 2, 2)
```

File: numpy/core/fromnumeric.py [hunk 4]
```diff
@@ -649,6 +649,14 @@ def transpose(a, axes=None):
     >>> np.transpose(x, (1, 0, 2)).shape
     (2, 1, 3)
 
+    >>> x = np.arange(120).reshape(2, 3, 4, 5)
+    >>> x.shape
+    (2, 3, 4, 5)
+
```
Comment (eric-wieser) [inline numpy/core/fromnumeric.py:655]: These lines aren't really interesting, the shape of the array should be obvious from the line above

File: numpy/core/fromnumeric.py [hunk 5]
```diff
@@ -649,6 +649,14 @@ def transpose(a, axes=None):
     >>> np.transpose(x, (1, 0, 2)).shape
     (2, 1, 3)
 
+    >>> x = np.arange(120).reshape(2, 3, 4, 5)
```
Comment (eric-wieser) [inline numpy/core/fromnumeric.py:652]: Since you're not showing the valeus here, may as well match the line above:

```suggestion
    >>> x = np.ones((2, 3, 4, 5))
```

File: numpy/core/fromnumeric.py [hunk 6]
```diff
@@ -649,6 +649,14 @@ def transpose(a, axes=None):
...
+    >>> x.shape
+    (2, 3, 4, 5)
+
+    >>> x_transposed = np.transpose(x)
+    >>> x_transposed.shape
+    (5, 4, 3, 2)
```
Comment (eric-wieser) [inline numpy/core/fromnumeric.py:658]: For consistency with the example above, I'd combine these lines:

```suggestion
    >>> np.transpose(x).shape
    (5, 4, 3, 2)
```

**Comment by ruppysuppy** [comment]
@eric-wieser is it now acceptable?

**Review by eric-wieser — APPROVED** [review]

**Comment by mattip** [comment]
Thanks @ruppysuppy

**Comment by ruppysuppy** [comment]
@mattip, the pleasure was all mine. And thanks a lot to @eric-wieser too, for constantly helping me out :)
